components/profile/user-name-form/view.tsx

```diff
@@ -0,0 +1,83 @@
+import React from "react";
+
+import { useTranslation } from "next-i18next";
+
+import Button from "components/button";
+import If from "components/If";
+import UserNameWithEditIcon from "components/profile/user-name-with-edit-icon/view";
+import ResponsiveWrapper from "components/responsive-wrapper";
+
+interface UserNameFormProps {
+  userName: string;
+  isEditUserName: boolean;
+  isSaveButtonDisabled: boolean;
+  isEmailInvalid: boolean;
+  isExecuting: boolean;
+  onHandleUserNameChange: (e) => void;
+  onHandleEditUserName: (e: boolean) => void;
+  onSave: () => void;
+}
+
+export default function UserNameForm({
```

Comment:
this view is missing a controller, profile-page should not be the controller for this view